events/UserDataCollectionEvent.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\legal\events;
+
+use humhub\modules\comment\models\Comment;
+use humhub\modules\file\models\File;
+use humhub\modules\like\models\Like;
+use humhub\modules\post\models\Post;
+use humhub\modules\rest\definitions\CommentDefinitions;
+use humhub\modules\rest\definitions\FileDefinitions;
+use humhub\modules\rest\definitions\LikeDefinitions;
+use humhub\modules\rest\definitions\PostDefinitions;
+use humhub\modules\rest\definitions\UserDefinitions;
+use humhub\modules\user\events\UserEvent;
+
+class UserDataCollectionEvent extends UserEvent
+{
+    public array $userData = [];
+
+    /**
+     * @inheritdoc
+     */
+    public function init()
+    {
+        $this->addUserData('user', UserDefinitions::getUser($this->user));
+
+        $this->addUserData('post', array_map(function ($post) {
+            return PostDefinitions::getPost($post);
+        }, Post::findAll(['created_by' => $this->user->id])));
+
+        $this->addUserData('comment', array_map(function ($comment) {
+            return CommentDefinitions::getComment($comment);
+        }, Comment::findAll(['created_by' => $this->user->id])));
+
+        $this->addUserData('file', array_map(function ($file) {
+            return FileDefinitions::getFile($file);
+        }, File::findAll(['created_by' => $this->user->id])));
+
+        $this->addUserData('like', array_map(function ($like) {
+            return LikeDefinitions::getLike($like);
+        }, Like::findAll(['created_by' => $this->user->id])));
+    }
+
+    public function addUserData(string $name, array $data)
```

Comment:
@yurabakhtin Can we just name it `addData`?  Since it's a UserEvent anyway...

Reply:
@luke- I thought about this too and I tried to use there a property name `data` instead of current `userData`, but the class `yii\base\Event` already has the property `public $data;` and it is reset to `null` after switch between triggers, i.e. if on init method we have the array filled with 5 default items then on go to next module I see the `$data` is empty again, so it is why I have decided to use new property with name `$userData` and also I have decided to use the the method name as `addUserData()` instead of `addData()` because I am afraid the method `addData()` may be used in future by the class `yii\base\Event` and we will may have a conflict in their declarations.

Reply:
@yurabakhtin Ah ok. Can we rename it to 

- `addExportData($category, $data)` 
- `addExportFile($fileName, $sourceFilePath)`

Reply:
@luke- Done in the next commit [here](https://github.com/humhub-contrib/legal/pull/77/commits/445058a081a9745607a5e1cc3560619e52908ae3#diff-48c316979c28746ad47d748873e08bf868ebe64219e89868e07a0b4fb9787c3bR74-R96).